<head>
   <title>Nucleotide Sequence Molecular Weight Calculator</title>
   <META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
   <LINK href="mainstyle.css" rel="styleSheet" type="text/css">
</head>
<body TEXT="#000000" bgcolor="#FFFFFF" LINK="#0000EF" VLINK="#CF30CF" ALINK="#FF0000">
<table border=0 cellpadding=5>
<tr>
<td>
<img src=ultrascan1.png border=0>
</td>
<td align=vcenter>
<h1 align=center><i>Manual</i></h1>
</td>
</tr>
</table>
<hr>
<p>
<center>
<H2>Nucleotide Sequence Molecular Weight Calculator:</H2>
<p>
<img src=nucleotide1.png>
</center>
<p>
This program allows you to calculate the molecular weight of DNA and
RNA molecules from sequence files which are written in the Genbank
format (text format). The program can be used to calculate single- or
double stranded sequences, or the molecular weight of the complement
of a sequence. You can select the 3' and 5' terminus chemistry, and add
common counterions in molar ratios to each residue.
<p>
<b>Note:</b> This program will utilize the database backend of
<b><i>UltraScan</i></b>. This functionality allows you to save certain
files and data in a database, among them the nucleotide files analyzed
with this module. If the database system is not supported or available
on your system, you will get a <a href=db-error1.png>warning message</a>
when opening this module.

<p>
You can download DNA/RNA sequences directly into the calculator from
Genbank. Clicking on the "<b>Download Sequence</b>" button will 
connect you to the Entrez database for <a href=download_nucleotide.html>
downloading</a> Genbank sequence files.
<p>
If you would like to import your own sequence, you need to make sure
that the file is in the proper Genbank format. 
<p>
Example sequence:
<pre>
DEFINITION  exon from: Gallus gallus egg white lysozyme gene, partial cds.
ORIGIN
   1 agtcccgctg tgtgtacgac actggcaaca tgaggtcttt gctaatcttg gtgctttgct
  61 tcctgcccct ggctgctctg gggaaagtct ttggacgatg tgagctggca gcggctatga
 121 agcgtcacgg acttgataac tatcggggat acagcctggg aaact
  //
</pre>
<p> 
The following four fields are required:
<p>
Field 1: "DEFINITION"<br>
This field has to be spelled in upper case and everything following
it on the same line will be printed as the sequence title in the program.
<p>
Field 2: "ORIGIN"<br>
This field has to be spelled in upper case and should be the only word in this
line. This word indicates that the <i>next</i> line will contain the beginning
of the actual sequence. 
<p>
Field 3: The actual sequence (multiline). The sequence elements (a, g, c, t, u)
can be spelled in upper or lower case, and can contain spaces. Each line with
sequence data needs to contain exactly one string, separated by white space, in
front of the actual sequence data. This is ordinarily the position of the 
first residue listed in that line.
<p>
Field 4: The sequence terminator "//". This terminator symbol should be
on a separate line following the last sequence line. Any information past
the terminator symbol is ignored.
<p>
Amy other information contained in the file is ignored.
<p>
The program uses the following algorithm (Ref.: Steve Hardies, University
of Texas Health Science Center at San Antonio, Dept. of Biochemistry):
<p>
<pre>
Natural isotope adjusted MWs

C	12.011
N	14.0067
O	15.9994
H	1.00794
P	30.9738

Adenine C5N5H5
Guanine C5N5OH5
Thymine C5N2O2H6
Cytosine C4N3OH5
Uracil  C4N2O2H4

ribose C5O5H10
deoxyribose C5O4H10

phosphate PO4

Adding the base to the ribose involves loss of 1 H2O 
Bonding the phosphate to either 5' or 3' OH causes loss of 1 HO

An internal ribonucleotide is then calculated as base+ribose+phosphate
-H2O - 2 OH.

MW of an internal ribonucleotide:

dA - 313.21
dC - 289.19
dU - 290.17
dG - 329.21
dT - 304.63

to make RNA, add 15.99 for each nucleotide
to make terminal 3' OH add 17.01
to make terminal 3' phosphate add 94.97
to make terminal 5' phosphate - no change
to make terminal 5' OH subtract 77.96

Does not account for the 2nd dissociable H of the terminal phosphate, which 
has a pK somewhere around neutrality. It would be 7.2 for a free phosphate.
</pre>
<p>
<b>Explanation for fields and buttons:</b>
<p>
<table border=1 cellspacing=5 cellpadding=5>
<tr>
<td>
<img src=nucleotide2.png align=center>
</td>
<td>
This group of functions allows you to select if the selected molecule is
to be calculated as an RNA or DNA molecule, if the molecular weight of the
sequence should be calculated including the molecular weight of the 
complement (in that case select <b>"Double Stranded"</b>), and what
groups are on the 3' and 5' termini of the sequence.
</td>
</tr>

<tr>
<td>
<img src=nucleotide3.png align=center>
</td>
<td>
If this option is selected, the complementary sequence will be generated, and the 
molecular weight of the complement only will be reported.
</td>
</tr>

<tr>
<td>
<img src=nucleotide4.png align=center>
</td>
<td>
By clicking on the arrow buttons (single arrow: steps of 0.01, double arrow: steps of 0.1)
you can select the molar ratio of counterions per <i>nucleotide</i> of the sequence. For
example, if you selected a sodium ratio of 0.01, one in a 100 nucleotides of the
sequence would be considered as having one sodium ion bound, and the weight of that 
ion would be added to the total molecular weight. If the sequence contained less than a 
hundred nucleotides, the corresponding fractional molecular weight of a sodium ion would
be added to the reported molecular weight.
</td>
</tr>

<tr>
<td>
<img src=nucleotide5.png align=center>
</td>
<td>
These fields show the first line of the sequence identification listed in the 
sequence file, as well as the molecular weight and sequence composition. If you
are loading a DNA sequence, but selected the <b>"RNA"</b> option above, an
<a href=nucleotide8.png>warning message</a> will be shown. An equivalent
<a href=nucleotide9.png>error message</a> is shown if you selected <b>"DNA"</b>
but loaded an RNA sequence.
</td>
</tr>

<tr>
<td>
<img src=nucleotide5a.png align=center>
</td>
<td>
<li><b>Select Investigator:</b> When searching for nucleotide sequences
from the database, you can limit your search to only those sequences
belonging to a specific investigator in the database. When you click
on this button, the <a href=tbl_investigator.html> investigator
selection dialog</a> will open up and allow you to select the desired
investigator. When the sequence is loaded from the database, this
option is disabled, since an investigator is already associated with the
selected sequence.  Also, when committing a sequence to the database an
investigator needs to be chosen to be associated with the sequence.

</td>
</tr>

<tr>
<td>
<img src=nucleotide6.png align=center>
</td>
<td>
<b>Sequence Loading:</b><br>
You can load sequences into this program by one of two methods:
<ol>
<li><b>Load Sequence from HD</b><br>
Use this button to load an ASCII sequence file from your harddrive. A file
dialog will open and allow you to search for *.seq files.
<p>
<li><b>Query Sequence from DB</b><br>
If you are using a database backend for <b><i>UltraScan</b></i> you can query
the database for nucleotide sequence files that have previously been committed
to the database. Sequence files available in the database will be listed in the listbox
once you click on this button. More information about various database functions is
available <a href=database.html>elsewhere</a>.
</ol>
</td>
</tr>

<tr>
<td>
<img src=nucleotide7.png align=center>
</td>
<td>
<li><b>Download Sequence:</b> Download a sequence from the <a href=download_nucleotide.html>
Genbank database</a>.<br>
<li><b>Save Sequence to DB:</b> Use this function to save a sequence to
the database.  Before storing a sequence in the database, you need to have
a copy of the sequence stored locally, for example, on your harddrive.
You can then select the sequence to be stored in the database from
a <a href=nucleotide11.png>filedialog</a> and after selection, <a
href=nucleotide12.png> commit</a> the sequence to the database.
<li><b>Calculate:</b> Update the molecular weight calculation and apply any ion
changes, if selected.
<li><b>Save Result:</b> Save the result of the calculation to a <a href=result.txt>file</a>.
This file will be saved in your <a href=nucleotide10.png>results directory</a>. The file
will be named by the name the sequence was stored, and a suffix of ".nucleotide_res" will
be added.
<li><b>Delete Sequence from DB:</b> Delete the selected sequence from the database.
This function requires <a href=administrator.html>administrator</a> privileges.
<li><b>Reset:</b> Clear the loaded sequence from memory
<li><b>Help:</b> Show this help file
<li><b>Close:</b> Close this function
</td>
</tr>

</table>
<hr>
<font face=helvetica size=-1>
www contact: <a href=mailto:demeler@biochem.uthscsa.edu>Borries Demeler</a>
<p>
This document is part of the <i><b>UltraScan</i></b> Software Documentation
distribution. <br><a href=copyright.html>Copyright &copy; notice</a>.<p> The latest version of this document can always be found at:
<p>
<ol>
<a href=http://www.ultrascan.uthscsa.edu>http://www.ultrascan.uthscsa.edu</a>
</ol>
<p>
Last modified on January 12, 2003.
</body>
</html>
