<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
   "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta content="text/html;charset=ISO-8859-1" http-equiv="Content-Type">
	<title>SOMO Miscellaneous Options Module</title>
	<link href="mainstyle3.css" rel="styleSheet" type="text/css">
</head>
<body>
<table BORDER=0 CELLPADDING=5 >
<tr>
<td><img SRC="USSOMO-banner.png" BORDER="0" ALT="UltraScan Version"> </td>
<td>
<center>
<h1>
<i>Manual</i></h1></center>
</td>
</tr>
</table>

<hr>
<center>
<h2>
SOMO - Miscellaneous Options Module:</h2>
<p>
<h3><center>Last updated: January 2023 </center></h3><p>
<p>
<img SRC="somo-misc1New.png" ALT="SOMO Miscellaneous Options Screen">
</center>
<p>

In this module, you can set some options affecting both the model building and the hydrodynamic computations stages. The <i><b>Calculate vbar</b></i> checkbox allows the computation of the <i>partial specific volume (vbar)</i> in standard
conditions (water @ 20&deg;C) of the structure from its composition, using the residues' values stored in the <a href= "somo-residue-0.png">residue table</a> (default: <b>active</b>).<br>
The <b><i>Covolume [cm^3/mol]:</i></b> field controls the addition of a covolume term (see Durchschlag and Zipper, Progr. Colloid. Polym. Sci. 94:20-39, 1994) to the calculation of the molar volume of an entire PDB structure, independently of how many chains it is composed from. Since this term was first added in the calculation of relatively small molecules (e.g., amino acids), and, as stated in the Durchschlag and Zipper paper quoted above "In the case of polymers the contribution of Vcv has to be neglected if the volume of the monomeric unit is to be calculated", we chose to add it just once per protein structure, although this is a matter of debate. (<b>default</b>: <b>12.4 cm<sup>3</sup>/mol</b>)<br>

Alternatively, a <i>vbar</i> value (<i>vbar(ent)</i>, either computed by other means or measured) can be entered in the <i><b>Enter a vbar value (cm^3/g)</b></i> field. For instance, if a significant number of non-coded or incomplete residues are skipped when parsing the PDB file, the computed <i>vbar</i> could be not correct; likewise, if the <a href=somo_pdb_parsing.html><b>Automatic Bead Builder </b></a> is used for non-coded residues, the <i>vbar</i> could also be incorrect. In these cases, entering an experimental value will be the best option. Importantly, the temperature <i>T(ent)</i> at which the entered <i>vbar(ent)</i> has been measured/computed must be entered in the <b><i>Vbar measured/computed at T=&deg;C:</i></b> field. The <i>vbar(ent)</i> and <i>T(ent)</i> values will be then used by the <a href=somo_hydro.html><b><i>Hydrodynamic Computation Options</i></b></a> module to calculate an operational <i>vbar</i> valid for the temperature conditions to which the hydrodynamic computations are referred. If a <i>vbar</i> value is entered in this field (and the <i><b>Calculate vbar</b></i> checkbox is deselected), a message will be displayed in the progress window ("ATTENTION: vbar = ") when the model is being built. This should avoid the use of incorrect external <i>vbar</i>
values resulting by inadvertently leaving the <i><b>Calculate vbar</b></i> checkbox deselected from a previous model-generating session. <br>The <i><b>Select vbar</b></i> button will instead load a <i>vbar</i> value calculated from protein sequence by using the <a href=vbar.html>UltraScan vbar calculation routine</a>. WARNING: whenever possible, use <b>experimental</b> <i>vbar</i> values, as many factors can affect its computation!!!! The routines provided here should be
considered just as a best approximation in absence of experimental data.
<p>
The <i><b>Hydration Water Vol. (A^3)</b></i> field sets the volume of the water of hydration molecules, which has been found (Gerstein and Chothia, Proc. Natl. Acad. Sci. USA 93:10167-10172, 1996) to be on average 24.5 &Aring;<sup>3</sup>,
different from that of either bulk (29.7 &Aring;<sup>3</sup>) or isolated (11.5 &Aring;<sup>3</sup>) water molecules (<b>default</b>: <b>24.041 &Aring;<sup>3</sup></b>).
<p>
The <i><b>Enable Peptide Bond Rule</b></i> checkbox controls if the <i>peptide bond rule</i> is used by the SoMo method. With this rule, the peptide bond segment is used for the main chain beads of a protein structure. These beads are thus positioned at the cog of the (CA-C-O)<sub>n</sub>-(N)<sub>(n+1)</sub> atoms, except when PRO is the (n+1) residue. In this case, the peptide bond bead is positioned at the cog of the (CA-C-O)<sub>n</sub> atoms. Additional rules control the generation of the OXT bead and of the first N atom at the beginning of each protein chain. All these rules are controlled by "special" residues in the <i>somo.residue</i> table. To gain total control over the positioning, volumes and masses
of every bead, the <i><b>Enable Peptide Bond Rule</b></i> checkbox should be <b>deselected</b>. In this case, the descriptions present in the <i>somo.residue</i> table are fully effective. (<b>default</b>: <b>selected</b>). 
<p>
The <b>Average Parameters for Automatic Bead Builder:</b> submenu contains a series of fields governing the <i>average parameters</i> that will be used by the <i>Automatic Bead Builder</i> routine when this option is selected (see
<a href=somo_pdb_parsing.html>here</a>) for non-coded residues.<br> 

<ul>
<li>
<b><i>Average atomic radius (A):</i></b> this is the radius used by the ASA routines (see <a href=somo_asa.html>here</a>) to compute the <i>accessible surface area</i> of the non-coded residue. The <b>default</b> value <b>(1.68 &Aring;)</b> was derived by averaging the atomic radii of the C,O,P,N,S elements present in the latest (January 2023) version of the default <i>somo.atom</i> table. Averaging over <b>all</b> elements produced a slightly lower value, 1.63 &Aring;.

<li>
<b><i>Average atomic mass (Da):</i></b> this is an average atomic mass for atoms within non-coded residues used in the computation of the total mass of the model. The <b>default</b> value <b>(14.5 Da)</b> was derived by averaging the atomic masses of the C,O,P,N,S elements present in the the latest (January 2023) version of the default <i>somo.atom</i> table (with implicit hydrogens counted). Including <b>all</b> elements produced a slightly larger value, 16.4 Da. In any case, we strongly suggest to enter a global value in the <i>Total Mass of Model</i> field in the <a href=somo_hydro.html><b><i>SOMO Hydrodynamic Calculation Options</i></b></a> panel when dealing with structures containing non-coded residues.

<li>
<b><i>Average atomic number of electrons:</i></b> used for computing the isoelectric point and net charge as a function of pH, and in the <i>P</i>(<i>r</i>) vs. <i>r</i> computations in the SAXS-framework. As for the two entries above, this average (7.7) was derived from the C,O,P,N,S elements present in the the latest (January 2023) version of the default <i>somo.atom</i> table. This includes the charged and not-charged species for ionizable groups (<i>e.g.</i>, -OH and -O<sup>-</sup>). Counting all elements produced a 8.5 value.

<li>
<b><i>Average atomic number of protons:</i></b> used for computing the isoelectric point and net charge as a function of pH. As above, this average (7.7) was derived from the C,O,P,N,S elements present in the the latest (January 2023) version of the default <i>somo.atom</i> table. This included the charged and not-charged species for ionizable groups (<i>e.g.</i>, -NH2 and -NH3<sup>+</sup>). Counting all elements produced a 8.5 value.

<li>
<b><i>Average atomic hydration:</i></b> this value will be multiplied by the number of atoms present in the non-coded residue to determine an hydration number (which will be rounded to the closest integer). The volume calculated from this hydration number multiplied by the <i>hydration water volume</i> (see above) will be added to the anhydrous bead volume (see below) to determine the final volume of the bead. The <b>default</b> value <b>(0.4)</b> is a compromise between the value derived for amino acids and carbohydrates (0.3) and that for nucleotides/nucleosides and prosthetic groups, 0.5. These number were calculated by averaging the hydration number of a bead divided by the number of atoms assigned to it for the beads defined in the original (April 2009) version of the default <i>somo.residue</i> table. This value should perhaps be adapted to the kind of non-coded residue(s) being roughly modeled by the <i>Automatic Bead Builder</i>.

<li>
<b><i>Average bead/atom volume:</i></b> this value will be multiplied by the number of atoms present in the non-coded residue to determine the anhydrous volume of the bead that will represent them, to which the volume of the water of hydration will be then added (see above). The <b>default</b> value <b>(15.3 &Aring;<sup>3</sup>)</b>  was derived by averaging the bead volume divided by the number of atoms assigned to it for each bead present in the original (April 2009) version of the default <i>somo.residue</i> table. Volumes by categories are: 17.8 &Aring;<sup>3</sup> for amino acids,
15.3 &Aring;<sup>3</sup> for carbohdrates, 14.7 &Aring;<sup>3</sup> for nucleotides/nucleosides, and 11.8 &Aring;<sup>3</sup> for prosthetic groups.

<li>
<b><i>Average Residue vbar:</i></b> this value will be used in the computation of the global <i>vbar</i> of the model. The <b>default</b> value <b>(0.72 cm<sup>3</sup>/g)</b> is that of the average protein. It could be changed when dealing with other kind of biomacromolecules. In any case, we strongly suggest to enter a global (better if experimental) value for <i>vbar</i> in the <b><i>Enter vbar:</i></b> field above when dealing with structures containing non-coded residues.

</ul>
<p>
The <b>Bead model controls</b> are user-defined parameters that were introduced in April 2012 and still need further testing, used when loading a bead model for SAXS computations.

<ul>
<li>
<b><i>Target electron density (A<sup>-3</sup>):</b></i> 
Set this value to the electron density of the structure.  

<li>
<b><i>Target volume (A<sup>3</sup>):</b></i> 
Set this value to the volume of the structure.

<li>
<b><i>Set targets on load PDB:</b></i> 
When set, loading a PDB will change the radii of the beads to match the <i>target electron density</i> and
<i>target volume</i> defined above.

<li>
<b><i>Equalize radii (constant volume):</b></i>
When set, all beads will be forced to have the same radius value.

</ul>
<p><p>
<b>Other Options:</b><p>
In May 2014 we added a new method for the computation of the hydrodynamic parameters starting from an atomic-level structure, the boundary elements approach <b><i>BEST</i></b> [S.R. Aragon, A precise boundary element method for macromolecular transport properties. J. Comp.Chem., 25, 1191-1205 (2004); S.R. Aragon and D.K. Hahn, Precise boundary element computation of proteins transport properties: Diffusion tensors, specific volume and hydration, Biophysical Journal, 91:1591-1603 (2006)]. This method necessitates a properly formatted atomic radii and names file for its ancillary program <b><i>MSROLL</i></b>, and the one provided with the original <b><i>BEST</i></b> program is directly made available within the <b>US-SOMO</b> implementation ("<i>best.radii</i>"). Users wishing to utilize instead the atomic radii and names as listed in the <b><i>US-SOMO</i></b> <i>somo.residue</i> file can generate a properly formatted file compatible with <b><i>BEST</i></b> by selecting the <i><b>Create MSROLL atomic radii and names file on load residue file</i></b> checkbox here, and re-launch <b><i>US-SOMO</i></b>. 

<hr>
<font face="helvetica" size="-1">
www contact: <a href="mailto:brookes@uthscsa.edu">Emre Brookes</a>
</font>
<p>
<font face="helvetica" size="-1">
This document is part of the <i><b>UltraScan</b></i> Software Documentation
distribution. <br><a href=copyright.html>Copyright &copy; notice</a>.<br><br> The latest version of this document can always be found at:
</font>
<p>
<font face="helvetica" size="-1">
<a href="http://somo.aucsolutions.com">http://somo.aucsolutions.com</a>
</font>
<p>
<font face="helvetica" size="-1">
Last modified on January 12, 2023.
</font>
</body>
</html>
