<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
   "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta content="text/html;charset=ISO-8859-1" http-equiv="Content-Type">
	<title>SOMO Miscellaneous Options Module</title>
	<link href="mainstyle.css" rel="styleSheet" type="text/css">
</head>
<body>
<table BORDER=0 CELLPADDING=5 >
<tr>
<td><img SRC="ultrascan1.png" BORDER="0" ALT="UltraScan Version"> </td>
<td>
<center>
<h1>
<i>Manual</i></h1></center>
</td>
</tr>
</table>

<hr>
<center>
<h2>
SOMO - Miscellaneous Options Module:</h2>
<p>
<img SRC="somo-misc1.png" ALT="SOMO Miscellaneous Options Screen">
</center>
<p>

In this module, you can set some options affecting both the model building and
the hydrodynamic computations stages. The <i><b>Calculate vbar</b></i> checkbox
allows the computation of the <i>partial specific volume (vbar)</i> in standard
conditions (water @ 20°C) of the structure from its composition, using the
residues' values stored in the <a href= "somo-residue-0.png">residue table</a>
(default: <b>active</b>). Alternatively, a <i>vbar</i> value (either computed by
other means or measured) can be entered in the <i><b>Enter a vbar value (cm^3/g)</b></i>
field. If a significant number of non-coded or incomplete residues are skipped when
parsing the PDB file, the computed <i>vbar</i> could be not correct; likewise, if
the <a href=somo_pdb_parsing.html><b>Automatic Bead Builder </b></a> is used for
non-coded residues, the <i>vbar</i> could also be incorrect. In these cases, entering
an experimental value will be the best option.
If a <i>vbar</i> value is entered in this field (and the <i><b>Calculate vbar</b></i> checkbox
is deselected), a message will be displayed in the progress window ("ATTENTION: vbar = ")
when the model is being built. This should avoid the use of incorrect external <i>vbar</i>
values resulting by inadvertently leaving the <i><b>Calculate vbar</b></i> checkbox deselected
from a previous model-generating session. <br>The <i><b>Select vbar</b></i> button will instead load
a <i>vbar</i> value calculated from protein sequence by using the <a href=vbar.html>UltraScan
vbar calculation routine</a>. WARNING: whenever possible, use <b>experimental</b> <i>vbar</i>
values, as many factors can affect its computation!!!! The routines provided here should be
considered just as a best approximation in absence of experimental data.

<p>The <i><b>Hydration Water Vol. (A^3)</b></i> field sets the volume of the water
of hydration molecules, which has been found (Gerstein and Chothia, Proc. Natl.
Acad. Sci. USA 93:10167-10172, 1996) to be on average 24.5 A<sup>3</sup>,
different from that of either bulk (29.7 A<sup>3</sup>) or isolated (11.5
A<sup>3</sup>) water molecules (default: <b>24.041 A<sup>3</sup></b>).
<p>
The <i><b>Enable Peptide Bond Rule</b></i> checkbox controls if the <i>peptide
bond rule</i> is used by the SoMo method. With this rule, the peptide bond segment is
used for the main chain beads of a protein structure. These beads are thus positioned at
the cog of the (CA-C-O)<sub>n</sub>-(N)<sub>(n+1)</sub> atoms, except when PRO is the (n+1)
residue. In this case, the peptide bond bead is positioned at the cog of the (CA-C-O)<sub>n</sub>
atoms. Additional rules control the generation of the OXT bead and of the first N atom at the
beginning of each protein chain. All these rules are controlled by "special" residues
in the <i>somo.residue</i> table. To gain total control over the positioning, volumes and masses
of every bead, the <i><b>Enable Peptide Bond Rule</b></i> checkbox should be <b>deselected</b>.
In this case, the descriptions present in the <i>somo.residue</i> table are fully effective.
(default: <b>selected</b>). 
<p>
The <b>Average Parameters for Automatic Bead Builder:</b> submenu contains a
series of fields governing the <i>average parameters</i> that will be used by the
<i>Automatic Bead Builder</i> routine when this option is selected (see
<a href=somo_pdb_parsing.html>here</a>) for non-coded residues.<br> 

<ul>
<li>
<b><i>Average atomic radius (A):</i></b>. This is the radius used by the ASA routines
(see <a href=somo_asa.html>here</a>) to compute the <i>accessible surface area</i>
of the non-coded residue. The <b>default</b> value <b>(1.68 A)</b>  was derived by averaging
the atomic radii of the C,O,P,N,S elements present in the current (April 2009) version of the default <i>somo.atom</i> table. Averaging over <b>all</b> elements produced essentially the same value, 1.67 A.

<li>
<b><i>Average atomic mass (Da):</i></b>. This is an average atomic mass for atoms within
non-coded residues used in the computation of the total mass of the model. The <b>default</b>
value <b>(16 Da)</b> was derived by averaging the atomic masses of the C,O,P,N,S elements present in the current (April 2009) version of the default <i>somo.atom</i> table. Including <b>all</b> elements produced a slightly larger value, 17.6 Da. In any case, we strongly suggest to enter a global value in the <i>Total Mass of Model</i> field in the <a href=somo_hydro.html><b><i>SOMO Hydrodynamic Calculation Options</i></b></a> panel when dealing with structures containing non-coded residues.


<li>
<b><i>Average atomic hydration:</i></b>. This value will be multiplied by the number of atoms
present in the non-coded residue to determine an hydration number (which will be rounded to the
closest integer). The volume calculated from this hydration number multiplied by the <i>hydration
water volume</i> (see above) will be added to the anhydrous bead volume (see below) to determine the final
volume of the bead. The <b>default</b> value <b>(0.4)</b> is a compromise between the value derived
for amino acids and carbohydrates (0.3) and that for nucleotides/nucleosides and prosthetic groups, 0.5.
These number were calculated by averaging the hydration number of a bead divided by the number of atoms
assigned to it for the beads defined in the current (April 2009) version of the default <i>somo.residue</i> table.  This value should perhaps be adapted to the kind of non-coded residue(s) being roughly modeled by the <i>Automatic Bead Builder</i>.

<li>
<b><i>Average bead/atom volume:</i></b>. This value will be multiplied by the number of atoms
present in the non-coded residue to determine the anhydrous volume of the bead that will
represent them, to which the volume of the water of hydration will be then added (see above).
The <b>default</b> value <b>(15.3 A<sup>3</sup>)</b>  was derived by averaging the bead volume 
divided by the number of atoms assigned to it for each bead present in the current (April 2009) version of the default <i>somo.residue</i> table. Volumes by categories are: 17.8 A<sup>3</sup> for amino acids,
15.3 A<sup>3</sup> for carbohdrates, 14.7 A<sup>3</sup> for nucleotides/nucleosides, and 11.8
A<sup>3</sup> for prosthetic groups.

<li>
<b><i>Average Residue vbar:</i></b>. This value will be used in the computation of the global 
<i>vbar</i> of the model. The <b>default</b> value <b>(0.72 cm<sup>3</sup>/g)</b> is that of
the average protein. It could be changed when dealing with other kind of biomacromolecules. In
any case, we strongly suggest to enter a global (better if experimental) value for <i>vbar</i> 
in the <b><i>Enter vbar:</i></b> field above when dealing with structures containing non-coded
residues.

</ul>
<hr>
<font face="helvetica" size="-1">
www contact: <a href="mailto:demeler@biochem.uthscsa.edu">Borries Demeler</a>
</font>
<p>
<font face="helvetica" size="-1">
This document is part of the <i><b>UltraScan</b></i> Software Documentation
distribution. <br><a href=copyright.html>Copyright &copy; notice</a>.<br><br> The latest version of this document can always be found at:
</font>
<p>
<font face="helvetica" size="-1">
<a href="http://www.ultrascan.uthscsa.edu">http://www.ultrascan.uthscsa.edu</a>
</font>
<p>
<font face="helvetica" size="-1">
Last modified on September 3, 2009.
</font>
</body>
</html>
